qhbmlib/inference/ebm.py

```diff
@@ -343,35 +343,74 @@ def _inner_log_partition():
     return _inner_log_partition()
 
   def _log_partition_forward_pass(self):
-    """Returns approximation to the log partition function.
-
-    Note this simple estimator is biased.  See the paper appendix for its
-    motivation.  TODO(#216)
+    r"""Returns approximation to the log partition function.
+
+    The calculation uses the uniform distribution to approximate the
+    partition function using Monte Carlo integration.  See section 11.2 of [1]
+    for details on Monte Carlo integration.
+
+    TODO (#216): decrease variance of this estimator.
+
+    Given an energy function $E$, the associated partition function is defined
+    $$
+    Z = \sum_x e^{-E(x)},
+    $$
+    where $x$ ranges over the domain of $E$.  We do not want to compute this sum
+    directly because it is over a set exponentially large in the number of bits.
+    Instead, we can use Monte Carlo integration.  To support this, consider
+    rewriting the sum as an expectation value with respect to the uniform
+    distribution $u(x)$:
+    $$
+    Z = \sum_x e^{-E(x)}
+      = \sum_x u(x) 2^n e^{-E(x)}
```

Comment:
I would pull 2^n out of the sum and expectation for better readability / minor clarity when taking the log

Reply:
Done, except on this first line, where `u(x) 2^n` is introducing identity